### ISSUE TITLE
Sensor Data in Homeassistant marked as unavailable most of the time

### ISSUE THREAD

**Issue by Eico-ccm** [alengwenus/esphome_components#12]
Hello,

I just wanted to know if my esp is set up correctly and if not how to fix it.

Currently the Data only shows up in Homeassistant when the smart meter gives out it's information. So there's always a gap in the Data that Homeassistant saves between the information bursts. When not recieving Homeassistant shows the sensor as not available.

But on the contrary the built-in esphome webserver component always shows the last values.

So why are the sensors labeled as unavailable most of the time?

Heres's the esp config:
[nmcu-esp32-berry-01.yaml.txt](https://github.com/alengwenus/esphome_components/files/9648962/nmcu-esp32-berry-01.yaml.txt)

Thank you for your time and this great component.

**Comment by alengwenus** [comment]
Hi @Eico-ccm,

sorry for the late reply.
Could you please try to set the text sensor's `format` parameter from `text` to `hex`?

**Comment by Eico-ccm** [comment]
Hey,

I just finished my setup. The Esp's Power is no longer a Trip hazard and the connections are soldered.

I made some config changes, the biggest was that i changed the esp-board type but the issue continued at first.
Then I let it run overnight and now the issue is gone?
So it either needed to run warm or the daily restart of my HASS-Instance fixed it somehow.

Anyway heres my setup for closeness sake.
[nmcu-esp32-berry-01.yaml.txt](https://github.com/alengwenus/esphome_components/files/12385698/nmcu-esp32-berry-01.yaml.txt)

Have a nice Day!
